src/v/kafka/client/producer.cc

```diff
@@ -144,12 +185,18 @@ producer::send(model::topic_partition tp, model::record_batch&& batch) {
       "send record_batch: {}, {{record_count: {}}}",
       tp,
       record_count);
-    auto p_id = tp.partition;
+    const auto p_id = tp.partition;
+    const auto auto_partition_swap_mitigation
+      = _config.produce_partition_agnostic_retries() > 0;
+    const auto retries = auto_partition_swap_mitigation
+                           ? _config.produce_partition_agnostic_retries()
+                           : _config.retries();
+    auto shared = batch.share();
     return ss::do_with(
              std::move(batch),
-             [this, tp](model::record_batch& batch) mutable {
+             [this, tp, retries](model::record_batch& batch) mutable {
                  return retry_with_mitigation(
-                   _config.retries(),
+                   retries,
```

Comment:
I believe there is an issue here. For auditing its desired that we have a high retry count for all instances `except` when the error code is `unknown_topic_or_partition`.

Reply:
I think it might be useful to extract a `produce_strategy` that:
1. Chooses a partition
2. "Sends" the batch
3. Handles the error

Hopefully the first case can be captured by moving when the `partitioner` is called and tweaking the 

`random_id_except` looks a bit like a `partitioner`.

The existing `partitioner` is `default_partitioner`
The existing error handling strategy is effectively `retry_with_mitigation`.

Reply:
Awesome thanks for the insight, sgtm